bazel/external/zlib.BUILD

```diff
@@ -0,0 +1,23 @@
+licenses(["notice"])  # Apache 2
+
+load("@envoy//bazel:envoy_build_system.bzl", "envoy_cmake_external", "envoy_package")
+
+envoy_package()
+
+envoy_cmake_external(
+    name = "zlib",
+    cache_entries = {
+        "BUILD_SHARED_LIBS": "off",
+        "CMAKE_CXX_COMPILER_FORCED": "on",
+    },
+    lib_source = "//:all",
+    static_libraries = select({
+        "@envoy//bazel:windows_x86_64": ["zlibstatic.lib"],
+        "//conditions:default": ["libz.a"],
+    }),
+)
```

Comment:
Why the movement? Is this the direction we want to take all the `foreign_cc` or is there a specific protobuf related reason to do this?

Reply:
I missed the last protobuf patch and revert this. but I think in future we might need this direction to remove `bind`.